python-package/lightgbm/dask.py

```diff
@@ -11,13 +11,13 @@
 from urllib.parse import urlparse
 
 import numpy as np
-import pandas as pd
 from dask import array as da
 from dask import dataframe as dd
```

Comment:
If you do this, then you need to make `dask.dataframe` a conditional import as well.

`pandas` is a required dependency of dask dataframe, so we can't require dask dataframe without requiring pandas. 

Try this

```shell
pip install --upgrade dask
pip uninstall -y pandas
python -c "from dask import dataframe as dd"
```

You'll get this error

```text
Traceback (most recent call last):
  File "/home/jlamb/miniconda3/lib/python3.7/site-packages/dask/dataframe/__init__.py", line 2, in <module>
    from .core import (
  File "/home/jlamb/miniconda3/lib/python3.7/site-packages/dask/dataframe/core.py", line 11, in <module>
    import pandas as pd
ModuleNotFoundError: No module named 'pandas'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/jlamb/miniconda3/lib/python3.7/site-packages/dask/dataframe/__init__.py", line 57, in <module>
    raise ImportError(msg) from e
ImportError: Dask dataframe requirements are not installed.

Please either conda or pip install as follows:

  conda install dask                     # either conda install
  python -m pip install "dask[dataframe]" --upgrade  # or python -m pip install
```

I don't think it's problematic to say that `pandas` is required to use the Dask interface.

Reply:
Ah, I didn't know that it is a transitive dependency. Thanks for pointing to it!